mockbroker.go

```diff
@@ -90,7 +90,7 @@ func (b *MockBroker) SetHandlerByMap(handlerMap map[string]MockResponse) {
 		fnMap[k] = v
 	}
 	b.setHandler(func(req *request) (res encoderWithHeader) {
-		reqTypeName := reflect.TypeOf(req.body).Elem().Name()
+		reqTypeName := reflect.TypeFor[protocolBody]().Name()
```

Comment:
This is not the same behavior. Because `protocolBody` is an interface, it gets passed into `reflect.TypeOf` without any conversion. Thus we’re getting the underlying type of what is _in_ `protocolBody` not that type itself. Here this would be a pointer to some concrete protocol body type, _not_ the `protocolBody` type itself.

The mistake here is that it is common that people will use `reflect.TypeOf(&interfacePrototype).Elem()` to get the type of the interface itself, since the dereference results in passing in a pointer-to-interface type into the `any` type parameter to `TypeOf`.

But that’s not the pattern we’re using here.

https://go.dev/play/p/ZwNsnoHSovl